pkg/serviceaccounts/controllers/pull_secrets.go

```diff
@@ -0,0 +1,42 @@
+package controllers
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+// findDockercfgSecret checks all the secrets in the namespace to see if the token secret has any existing dockercfg secrets that reference it
+func findDockercfgSecrets(client kubernetes.Interface, tokenSecret *corev1.Secret) ([]*corev1.Secret, error) {
+	dockercfgSecrets := []*corev1.Secret{}
+
+	options := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector(api.SecretTypeField, string(v1.SecretTypeDockercfg)).String()}
+	potentialSecrets, err := client.CoreV1().Secrets(tokenSecret.Namespace).List(options)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, currSecret := range potentialSecrets.Items {
+		if currSecret.Annotations[ServiceAccountTokenSecretNameKey] == tokenSecret.Name {
+			dockercfgSecrets = append(dockercfgSecrets, &potentialSecrets.Items[i])
+		}
+	}
+
+	return dockercfgSecrets, nil
+}
+
+// hasOwnerReference checks if the given tokenSecret is the owner of the provided pullSecret
+func hasOwnerReference(tokenSecret, pullSecret *v1.Secret) bool {
+	for _, owner := range pullSecret.OwnerReferences {
+		if owner.Kind == "Secret" &&
+			owner.APIVersion == "v1" &&
```

Comment:
@bparees I couldn't readily find constants in the kubernetes api that I could use here...

Reply:
ok

Reply:
@adambkaplan you can avoid using these constants by checking only UID, like it's done [in metav1.IsControlledBy](https://github.com/kubernetes/apimachinery/blob/79c2a76c473a20cdc4ce59cae4b72529b5d9d16b/pkg/apis/meta/v1/controller_ref.go#L29).